pkg/controllers/manifestwork.go

```diff
@@ -0,0 +1,250 @@
+	m := ScafoldManifestwork(hyd)
+
+	// if the manifestwork is created, then move the status to hypershiftDeployment
+	// TODO: @ianzhang366 might want to do some upate/patch when the manifestwork is created.
+	if err := r.Get(ctx, getManifestWorkKey(hyd), m); err == nil {
+		workConds := m.Status.Conditions
+
+		for _, cond := range workConds {
+			if err := r.updateStatusConditionsOnChange(
+				hyd,
+				hypdeployment.ConditionType(cond.Type),
+				metav1.ConditionTrue,
+				cond.Message,
+				cond.Reason,
+			); err != nil {
+				r.Log.Info(fmt.Sprintf("update status condition failed for %s%s, err: %v", getTargetNamespace(hyd), hyd.GetName(), err))
+				return ctrl.Result{RequeueAfter: 10 * time.Second, Requeue: true}, nil
+			}
+		}
+
+		return ctrl.Result{}, nil
+	}
+
+	appendSecrets, err := r.appendReferenceSecrets(ctx, hyd)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+	payload := []workv1.Manifest{}
+
+	manifestFuncs := []loadManifest{
+		appendHostedCluster,
+		appendNodePool,
+		appendSecrets,
+	}
+
+	for _, f := range manifestFuncs {
+		f(hyd, &payload)
+	}
+
+	m.Spec.Workload.Manifests = payload
+
+	// 	if err := controllerutil.SetOwnerReference(hyd, m, r.Scheme); err != nil {
+	// 		return ctrl.Result{}, fmt.Errorf("failed to set manifestwork's owner as %s, err: %w", req, err)
+	// 	}
+
+	if err := r.Create(r.ctx, m); err != nil {
+		if apierrors.IsAlreadyExists(err) {
+			//TODO: ianzhang366, might want to patch the manifestwork over here.
+			return ctrl.Result{}, r.Update(r.ctx, m)
+		}
+
+		return ctrl.Result{}, fmt.Errorf("failed to create manifestwork based on hypershiftDeployment: %s, err: %w", req, err)
+	}
+
+	r.Log.Info(fmt.Sprintf("created manifestwork for hypershiftDeployment: %s at targetNamespace: %s", req, getTargetNamespace(hyd)))
+
+	return ctrl.Result{}, nil
+}
+
+func (r *HypershiftDeploymentReconciler) deleteManifestworkWaitCleanUp(ctx context.Context, hyd *hypdeployment.HypershiftDeployment) (ctrl.Result, error) {
+
+	m := ScafoldManifestwork(hyd)
+
+	if err := r.Delete(ctx, m); err != nil {
+		if !apierrors.IsNotFound(err) {
+			return ctrl.Result{}, fmt.Errorf("failed to delete manifestwork, err: %v", err)
+		}
+	}
+
+	for _, np := range hyd.Spec.NodePools {
+		var nodePool hyp.NodePool
+		if err := r.Get(ctx, types.NamespacedName{Namespace: getTargetNamespace(hyd), Name: np.Name}, &nodePool); err == nil {
+			r.Log.Info(fmt.Sprintf("Waiting for NodePool %s/%s to be deleted", getTargetNamespace(hyd), np.Name))
+			return ctrl.Result{RequeueAfter: 10 * time.Second, Requeue: true}, nil
+		} else {
+			r.Log.Info(fmt.Sprintf("NodePool %s/%s already deleted...", getTargetNamespace(hyd), hyd.Name))
+		}
+	}
+
+	// Delete the HostedCluster
+	var hc hyp.HostedCluster
+	if err := r.Get(ctx, types.NamespacedName{Namespace: getTargetNamespace(hyd), Name: hyd.Name}, &hc); !apierrors.IsNotFound(err) {
```

Comment:
Same as above, does not apply to the case of one managed cluster as the hypershift management cluster.